### PR TITLE
[Improvement][API][num-1 & 2] Implement createEmptyProcessDefinition & updateProcessDefinitionBasicInfo

### PR DESCRIPTION
**Pull request by edward-yang**
#6772 

num | Interface | Interface method | Operate | Function description
-- | -- | -- | -- | --
1 | ProcessDefinition | createEmptyProcessDefinition | new | Create offline empty workflow and schedule at the same time. When schedulejson is empty, only empty work will be created
2 | ProcessDefinition | updateProcessDefinitionBasicInfo | new | Modify the basic information of workflow and schedule. When schedulejson is empty, only the basic information of workflow can be modified, or the online status can be specified. An empty workflow is not allowed to go online

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java [hunk 1]
```diff
@@ -1547,7 +1548,95 @@ public Result queryProcessDefinitionVersions(User loginUser, long projectCode, i
...
+    }
+
+    private Map<String, Object> createDagSchedule(User loginUser,
+                                                  long projectCode,
+                                                  long processDefinitionCode,
+                                                  String scheduleJson) {
+        Map<String, Object> result = new HashMap<>();
+        ScheduleDto scheduleDto = JSONUtils.parseObject(scheduleJson, ScheduleDto.class);
+        ScheduleParam schedule = scheduleDto.getSchedule();
+        FailureStrategy failureStrategy = scheduleDto.getFailureStrategy() == null ? FailureStrategy.CONTINUE : scheduleDto.getFailureStrategy(); // 默认：CONTINUE
```
Comment (brave-lee) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java]: please use English comments

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/ScheduleDto.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+package org.apache.dolphinscheduler.api.dto;
+
+import org.apache.dolphinscheduler.common.enums.FailureStrategy;
+import org.apache.dolphinscheduler.common.enums.Priority;
+import org.apache.dolphinscheduler.common.enums.WarningType;
+
+/**
+ * schedule dto
+ **/
+public class ScheduleDto {
```
Comment (brave-lee) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/ScheduleDto.java]: I think this is not necessary, you can use `Schedule` instead.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java (hunk 1 above)
Comment (edward-yang) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java]: I had replace class ScheduleDto with Schedule and remove it

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/ScheduleDto.java (hunk 2 above)
Comment (edward-yang) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/ScheduleDto.java]: I had replace class ScheduleDto with Schedule and remove it

**Review by brave-lee — DISMISSED** [review]
+1

**Review by brave-lee — APPROVED** [review]
+1

**Pull request merged**